base/float.jl

```diff
@@ -1064,6 +1064,11 @@ julia> 1.0 + eps()
 julia> 1.0 + eps()/2
 1.0
 ```
+
+For complex inputs `T`, `eps(T)` is defined as the distance bound to the nearest
+floating-point complex value, i.e. for ``z=a+ib`` and the closest floating-point
+complex value ``z̃=ã+ib̃=fl(a)+ifl(b)`` where ``fl(x)`` is the closest
+floating-point value to a real ``x``, ``eps(z)`` satisfies ``|z - z̃| ≤ eps(z)/2``.
```

Comment:
It would be good to frame this as a strict generalization of the definition, since it is also true for real numbers, and then give complex numbers as a special case.

```suggestion
More generally, for any floating-point numeric type, `eps` corresponds to an
upper bound on the distance to the nearest floating-point complex value: if ``\text{fl}(x)`` is the closest
floating-point value to a number ``x`` (e.g. an arbitrary real number), then ``\text{fl}(x)``
satisfies ``|z - \text{fl}(x)| ≤ \text{eps}(z)/2``. This allows the definition of `eps` to be
extended to complex numbers, for which ``\text{fl}(a + ib) = \text{fl}(a) + i \text{fl}(b)``.
```

(Note that `eps(x)` and `fl(x)` should ideally be typeset with the AMS `\mathoperator` so that they aren't italicized, but that isn't supported in all LaTeX-math environments, and so I used `\text` instead to get $\text{eps}(x)$ and $\text{fl}(x)$.)